babel/dates.py

```diff
@@ -1177,12 +1181,19 @@ def parse_date(string, locale=LC_TIME, format='medium'):
     #        names, both in the requested locale, and english
 
     year = numbers[indexes['Y']]
+    day = numbers[indexes['D']]
+    month = numbers[indexes['M']]
     if len(year) == 2:
-        year = 2000 + int(year)
-    else:
-        year = int(year)
-    month = int(numbers[indexes['M']])
-    day = int(numbers[indexes['D']])
+        # check if we don't have an ISO kind of format
```

Comment:
There's a comment on line 1168 with "TODO: try ISO format first" – maybe this logic should be around there. Also, I'm not sure it makes much sense to read any of the indexes from a localized format for the very much standardized ISO order...

Reply:
> There's a comment on line 1168 with "TODO: try ISO format first" – maybe this logic should be around there.

I can remove the comment, I just forgot.

> Also, I'm not sure it makes much sense to read any of the indexes from a localized format for the very much standardized ISO order...

I thought about this but then I thought that it's probably best to be more flexible and accept also formats which only look like ISO e.g. with slashes as separator rather than dashes.
This said, I don't care too much, I do use properly ISO :-) Just let me know.

Reply:
OK, I pushed a new commit with a more explicit handling of the ISO format at the beginning of the function. Let me know which one you prefer.